src/main/scala/net/snowflake/spark/snowflake/pushdowns/querygeneration/MiscStatement.scala

```diff
@@ -168,8 +171,16 @@ private[querygeneration] object MiscStatement {
   private final def setToExpr(set: Set[Any]): Seq[Expression] = {
     set.map {
       case d: Decimal => Literal(d, DecimalType(d.precision, d.scale))
-      case s: String => Literal(s, StringType)
+      case s @ (_:String | _:UTF8String) => Literal(s, StringType)
+      case d: Double => Literal(d, DoubleType)
       case e: Expression => e
+      case default =>
```

Comment:
1. Use format like: s"${default.getClass.toString} @ MiscStatement. setToExpr"
2. Add comment before Throw like:
            // This exception is not a real issue. It will be caught in
            // QueryBuilder.treeRoot and a telemetry message will be sent if
            // there are any snowflake tables in the query.